### ISSUE TITLE
Feature Request: Play/Pause Button

### ISSUE THREAD

**Issue by alexandergerber** [johnwalley/d3-simple-slider#74]
I would really like to have an optional play/pause button for the sliders. 
Would that be within the scope of your project?

**Comment by johnwalley** [comment]
I'd say it's outside the scope of the project. I see it being made up of two parts:

1. ui controls for play/pause
1. functionality to repeatedly update the value of the slider when it is in the 'play' state

That said, I think it would make a good example/demo. I've got no immediate plans to add any new examples but I will update this issue if I do get around to it.

If you wanted to have a go yourself, perhaps as a [codepen](https://codepen.io/), I'd be happy to provide assistance.

**Comment by geohuz** [comment]
I'm wondering except for setting value (slider.value(x)) to make the slider changes, is there any api to tell the slider step into next point according to the setup? (Eg, we are set min, max, and step, now we just call sth to make it forward or backward?)

**Comment by johnwalley** [comment]
Hi @geohuz and thanks for the question. You are right in saying that the only way to currently do this is to set the slider value explicitly. I don't have any current plans to add a method to to step to the next (or previous point).

However, I'll finally get around to putting together an example of doing this in the next week or so.

**Comment by johnwalley** [comment]
I've started on an example in this PR #127 

![Nov-11-2020 20-29-56](https://user-images.githubusercontent.com/981531/98861216-d390b780-245c-11eb-9281-ed3254c9703c.gif)

**Comment by geohuz** [comment]
This is awesome! Can we use it for now?

**Comment by johnwalley** [comment]
Yes, feel free to use the code. I'll also merge the PR this weekend.

**Comment by geohuz** [comment]
hi @johnwalley would you give me a link for the example  code? I could't find it any where, thanks

**Comment by johnwalley** [comment]
https://github.com/johnwalley/d3-simple-slider/pull/168/files

Still in the PR for now :)

**Comment by geohuz** [comment]
Hi @johnwalley  thanks for the demo, I've tried to wrap the player into react component, but I found the marker don't move, only the fill progress. Would you mind help me to fix the issue? Here is the link on code sandbox: https://codesandbox.io/s/optimistic-brook-l09pp?file=/src/App.js

Thanks

**Comment by johnwalley** [comment]
You could do something like this: https://codesandbox.io/s/angry-river-gldvt.

They key is to avoid getting into an infinite loop (user moves handle -> call setPosition -> set value on slider -> call setPosition ->...) which is where slientValue helps.

**Comment by geohuz** [comment]
@johnwalley appreciate it so much! problem solved!
